src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspace.cs

```diff
@@ -32,44 +34,57 @@ static VisualStudioWorkspace()
         internal VisualStudioWorkspace(HostServices hostServices)
             : base(hostServices, WorkspaceKind.Host)
         {
-            _backgroundCompiler = new BackgroundCompiler(this);
-
-            var cacheService = Services.GetService<IWorkspaceCacheService>();
-            if (cacheService != null)
+            _backgroundCompilationDisabled = this.Services.GetRequiredService<IWorkspaceConfigurationService>().Options.DisableBackgroundCompilation;
+            if (!_backgroundCompilationDisabled)
             {
-                cacheService.CacheFlushRequested += OnCacheFlushRequested;
-            }
+                _backgroundCompiler = new BackgroundCompiler(this);
 
-            _backgroundParser = new BackgroundParser(this);
-            _backgroundParser.Start();
-        }
+                var cacheService = Services.GetService<IWorkspaceCacheService>();
+                if (cacheService != null)
+                {
+                    cacheService.CacheFlushRequested += OnCacheFlushRequested;
+                }
 
-        private void OnCacheFlushRequested(object sender, EventArgs e)
-        {
-            if (_backgroundCompiler != null)
-            {
-                _backgroundCompiler.Dispose();
-                _backgroundCompiler = null; // PartialSemanticsEnabled will now return false
+                _backgroundParser = new BackgroundParser(this);
+                _backgroundParser.Start();
             }
 
-            // No longer need cache notifications
-            var cacheService = Services.GetService<IWorkspaceCacheService>();
-            if (cacheService != null)
+            return;
+
+            void OnCacheFlushRequested(object sender, EventArgs e)
             {
-                cacheService.CacheFlushRequested -= OnCacheFlushRequested;
+                if (_backgroundCompiler != null)
+                {
+                    _backgroundCompiler.Dispose();
+                    _backgroundCompiler = null; // PartialSemanticsEnabled will now return false
+                }
+
+                // No longer need cache notifications
+                var cacheService = Services.GetService<IWorkspaceCacheService>();
+                if (cacheService != null)
+                    cacheService.CacheFlushRequested -= OnCacheFlushRequested;
             }
         }
 
         protected internal override bool PartialSemanticsEnabled
         {
-            get { return _backgroundCompiler != null; }
+            get
+            {
+                // Still allow for partial semantics in the VS workspace, even with background compilation disabled. The
+                // presumption here is that there's *always* something implicitly pulling the compilations forward in VS
+                // (e.g. tagging, completion, etc.) at a regular cadence.
+                if (_backgroundCompilationDisabled)
```

Comment:
Would requesting a frozen partial solution also pulling the complilations forward?

Reply:
no.  frozen partial is specifically around the idea of saying "i want a snapshot of whatever you have now, i don't want to wait for all information to be up to date.  but make sure you at least have the contents of this documetn correct as i'm going to ask about the tree/semantics for *it*".

Reply:
ok, then we need to remove "completion" from the features that pull us forward in the comment :)

Reply:
fair.  will update :)